### PR TITLE
Prevent null element in schemas with ignored fields

### PR DESCRIPTION
**Pull request by cee-dub**
Hi, thanks for your library! I'm finding it very helpful.

I needed to include a helper object into a struct from which I was creating a schema. On this field, I added the tag `json:"-"` to allow both `encoding/json` and `bqschema` to ignore it. Unfortunately, the JSON output of the schema then includes a `null` in the `"fields"` array corresponding to this ignored field.

I fixed this behavior in `8fa16a0` and made a few minor cleanups in the following commit. (They prevent needing to import the package in the tests.)

Hoping you'll find these changes amenable.

Cheers,
**—cw**

### CODE REVIEW

File: bqschema_suite_test.go [hunk 1]
```diff
@@ -1,10 +1,10 @@
-package bqschema_test
+package bqschema
```
Comment (SeanDolphin) [inline bqschema_suite_test.go:1]: Why this change?  Changing the test package this way will give the test package the ability to see all private items instead of just the publish facing elements?

It could be rewritten as import( . "github.com/SeanDolphin/bqschema") if you did not want the explicient call to bqschema?

**Comment by SeanDolphin** [comment]
Beyond the question on the test package naming the rest of the code looks good.  Once we determine the answer to that question. I will gladly merge in this pull request.  Thanks for the pull request.

File: bqschema_suite_test.go (hunk 1 above)
Comment (cee-dub) [inline bqschema_suite_test.go:1]: All of what you say is true. I made this change because the usage of package `*_test` for tests has two specific purposes: avoiding an import cycle, and  black box testing. Neither of those is necessary here.

If there were a true black box test, I agree it should be in either `package bqschema_test` or another package.

Comment (SeanDolphin) [inline bqschema_suite_test.go:1]: In this case it does not change how the program acts or tests, I generally just prefer to have tests keep as packaged with the _test to make testing only call the Public functions. I feel it helps generate a more accurate test coverage number.  All of that is opinion not based on anything besides my feelings.  If you could change it to the _test package name or give a better reason than my I like it as _test package name, I will merge after that reason or change.  Thanks.

Comment (cee-dub) [inline bqschema_suite_test.go:1]: I don't feel strongly that the tests must be in `package bqschema` rather than `bqschema_test` so I'll change this to use the dot import style.

Comment (cee-dub) [inline bqschema_suite_test.go:1]: When trying to apply the change, I came up with another reason why it works better to have the tests _not_ use a dot import: I cannot produce a merge-able commit that you can just click the button on. I'm forced to commit with `import . "github.com/nbio/bqschema"` in my repo, and you would have to merge by hand, fixing the import path.

This complication does not arise when the tests are in `package bqschema`.

Comment (cee-dub) [inline bqschema_suite_test.go:1]: I have the above commit ready, but I chose not to push it pending further discussion.

**Comment by cee-dub** [comment]
Thanks for the merge! Your point about test coverage is interesting. I hadn't thought of that before.
